### PR TITLE
added program which do real coffee machine auctions

### PR DESCRIPTION
**Pull request by Edvard777**

### CODE REVIEW

File: main/src/CoffeeMachineSimulator.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    public void espresso() {
+        if (water >= 250 && coffeeBeans >= 16 && disCups != 0) {
+            System.out.println("I have enough resources, making you a coffee!");
+            water -= 250;
+            coffeeBeans -= 16;
+            money += 4;
+            disCups -= 1;
+            System.out.println("Write action (buy, fill, take, remaining, exit):");
+            action(CoffeeMachineSimulatorMain.getString());
+        } else if (water < 250) {
```
Comment (hhov88) [inline main/src/CoffeeMachineSimulator.java:133]: this part is the same in most of the function like espresso, cappuccino or latte, can you fix it ?

File: main/src/CoffeeMachineSimulator.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+                break;
+        }
+    }
+
+   // action method which check input, change machine mode and call appropriate method
+    public void action(String actionType) {
+
+        switch (actionType) {
+            case "buy":
+                mode = 2;
```
Comment (hhov88) [inline main/src/CoffeeMachineSimulator.java]: you set mode to 2, and after that in buy function its become a 1, what is the meaning of this ?

File: main/src/CoffeeMachineSimulatorMain.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+public class CoffeeMachineSimulatorMain {
+
+
+     public static String getCurrentLine;
+
+    //printing auctions, call myMethod
+    public static void main(String[] args) {
+        System.out.println("Write action (buy, fill, take, remaining, exit):");
+        getCurrentLine = getString();
```
Comment (hhov88) [inline main/src/CoffeeMachineSimulatorMain.java]: I think you need some loop to call getString function

File: main/src/CoffeeMachineSimulator.java [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    String fill = "fill";
+    String take = "take";
+    String exit = "exit";
+    String remaining = "remaining";
+    String back = "back";
+    int mode = 1;
+
+
+    // method myMethod check coffee machine mode, and call appropriate method
+    public void myMethod(String inputString) {
```
Comment (hhov88) [inline main/src/CoffeeMachineSimulator.java]: it's not good ida to set method name like this, try to set same meaning names checkMode for example

Comment (Edvard777) [reply main/src/CoffeeMachineSimulator.java]: done

File: main/src/CoffeeMachineSimulatorMain.java (hunk 3 above)
Comment (Edvard777) [reply main/src/CoffeeMachineSimulatorMain.java]: Hayk but all working

File: main/src/CoffeeMachineSimulator.java (hunk 2 above)
Comment (Edvard777) [reply main/src/CoffeeMachineSimulator.java]: it was been in a homework task, that we must store current state, in any case i think it not needed

File: main/src/CoffeeMachineSimulatorMain.java (hunk 3 above)
Comment (hhov88) [reply main/src/CoffeeMachineSimulatorMain.java]: I mean we need to create a program which is going to make latte espresso or whatever and finish when user set finish command yes ?

Comment (Edvard777) [reply main/src/CoffeeMachineSimulatorMain.java]: Hayk jan i have a action method which when it see input"exit" is finishing program

Comment (Edvard777) [reply main/src/CoffeeMachineSimulatorMain.java]: and in other cases too but, fill, remaining, e.t.c

File: main/src/CoffeeMachineSimulator.java (hunk 1 above)
Comment (Edvard777) [reply main/src/CoffeeMachineSimulator.java:133]: as i use class method in other class, i find this solution more effective.

File: main/src/CoffeeMachineSimulator.java [hunk 5]
```diff
@@ -0,0 +1,200 @@
+import java.util.Scanner;
+
+public class CoffeeMachineSimulator {
+
+    //declaring our class variables
```
Comment (GayaneGulanyan) [inline main/src/CoffeeMachineSimulator.java]: Is it necessary?

File: main/src/CoffeeMachineSimulator.java [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+    String fill = "fill";
+    String take = "take";
+    String exit = "exit";
+    String remaining = "remaining";
+    String back = "back";
+    int mode = 1;
+
+
+    // method myMethod check coffee machine mode, and call appropriate method
+    public void checkMethod(String inputString) {
```
Comment (nsimonyan) [inline main/src/CoffeeMachineSimulator.java]: @Edvard777 please change method name it is not descriptive , and the variable name as well . Also you can try to use java doc comments , write `/**` press enter comment will be generated . And in the /** **/ write your comment

File: main/src/CoffeeMachineSimulator.java (hunk 1 above)
Comment (nsimonyan) [reply main/src/CoffeeMachineSimulator.java:133]: @Edvard777 dublicate code is effective ?

File: main/src/CoffeeMachineSimulatorMain.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import java.util.Scanner;
+
+public class CoffeeMachineSimulatorMain {
+
+
+     public static String getCurrentLine;
```
Comment (nsimonyan) [inline main/src/CoffeeMachineSimulatorMain.java]: @Edvard777 why this is not in main method

File: main/src/CoffeeMachineSimulatorMain.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+        System.out.println("Write action (buy, fill, take, remaining, exit):");
+        getCurrentLine = getString();
+        CoffeeMachineSimulator coffeeMachineSimulator = new CoffeeMachineSimulator();
+        coffeeMachineSimulator.checkMethod(getCurrentLine);
+
+
+    }
+
+     // method getString read input from console
+    public static String getString() {
```
Comment (nsimonyan) [inline main/src/CoffeeMachineSimulatorMain.java]: @Edvard777 jan please rename not descriptive

File: main/src/CoffeeMachineSimulator.java [hunk 9]
```diff
@@ -0,0 +1,200 @@
+import java.util.Scanner;
+
+public class CoffeeMachineSimulator {
```
Comment (nsimonyan) [inline main/src/CoffeeMachineSimulator.java]: @Edvard777 jan please add a readme

**Review by nsimonyan — CHANGES_REQUESTED** [review]

File: main/src/CoffeeMachineSimulator.java (hunk 5 above)
Comment (Edvard777) [reply main/src/CoffeeMachineSimulator.java]: i think no

File: main/src/CoffeeMachineSimulator.java (hunk 6 above)
Comment (Edvard777) [reply main/src/CoffeeMachineSimulator.java]: done

File: main/src/CoffeeMachineSimulatorMain.java (hunk 7 above)
Comment (Edvard777) [reply main/src/CoffeeMachineSimulatorMain.java]: done

File: main/src/CoffeeMachineSimulatorMain.java (hunk 8 above)
Comment (Edvard777) [reply main/src/CoffeeMachineSimulatorMain.java]: done

File: main/src/CoffeeMachineSimulator.java (hunk 9 above)
Comment (Edvard777) [reply main/src/CoffeeMachineSimulator.java]: done

**Comment by GayaneGulanyan** [comment]
Approved from my side

**Review by GayaneGulanyan — COMMENTED** [review]
Approved from my side

File: main/src/CoffeeMachineSimulator.java [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+
+    int water = 400;
```
Comment (nsimonyan) [inline main/src/CoffeeMachineSimulator.java]: @Edvard777 why not private ?

File: main/src/CoffeeMachineSimulator.java [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+    int milk = 540;
+    int coffeeBeans = 120;
+    int disCups = 9;
+    int money = 550;
+    String buy = "buy";
+    String fill = "fill";
```
Comment (nsimonyan) [inline main/src/CoffeeMachineSimulator.java]: @Edvard777 you can try to use Enum

File: main/src/CoffeeMachineSimulator.java [hunk 12]
```diff
@@ -0,0 +1,199 @@
...
+    String take = "take";
+    String exit = "exit";
+    String remaining = "remaining";
+    String back = "back";
+    int mode = 1;
+
+
+    // method checkMethod check coffee machine mode, and call appropriate method
```
Comment (nsimonyan) [inline main/src/CoffeeMachineSimulator.java]: @Edvard777 please try to update comments as I show yesterday

File: main/src/CoffeeMachineSimulator.java (hunk 11 above)
Comment (Edvard777) [reply main/src/CoffeeMachineSimulator.java]: Naira jan bayc enum chenq ancel

File: main/src/CoffeeMachineSimulator.java (hunk 10 above)
Comment (Edvard777) [reply main/src/CoffeeMachineSimulator.java]: If about Class modifier, if private i can't call it from package.Can make Default modifier

File: main/src/CoffeeMachineSimulator.java (hunk 1 above)
Comment (Edvard777) [reply main/src/CoffeeMachineSimulator.java:133]: done, maked loop

File: main/src/CoffeeMachineSimulator.java (hunk 12 above)
Comment (Edvard777) [reply main/src/CoffeeMachineSimulator.java]: done

**Review by hhov88 — APPROVED** [review]

**Pull request merged**